kernel/src/main/java/org/kframework/krun/KRun.java

```diff
@@ -152,17 +153,22 @@ public static Rule parsePattern(FileUtil files, KExceptionManager kem, String pa
 
     private K parseConfigVars(KRunOptions options, CompiledDefinition compiledDef) {
         HashMap<KToken, K> output = new HashMap<>();
+        scala.collection.Set<KToken> expectedConfigVars = new ConfigurationInfoFromModule(compiledDef.kompiledDefinition.mainModule()).configVars();
         for (Map.Entry<String, Pair<String, String>> entry
                 : options.configurationCreation.configVars(compiledDef.getParsedDefinition().mainModule().name(), files).entrySet()) {
             String name = entry.getKey();
-            String value = entry.getValue().getLeft();
-            String parser = entry.getValue().getRight();
             String configVarName = "$" + name;
-            Sort sort = compiledDef.configurationVariableDefaultSorts.getOrDefault(configVarName, compiledDef.programStartSymbol);
-            K configVar = externalParse(parser, value, sort, Source.apply("<command line: -c" + name + ">"), compiledDef);
-            output.put(KToken(configVarName, Sorts.KConfigVar()), configVar);
+            if (!expectedConfigVars.contains(KToken(configVarName, Sorts.KConfigVar())) && !name.equals("$STDIN") && !name.equals("$IO")) {
+                kem.registerCompilerWarning(ExceptionType.INVALID_CONFIG_VAR, "Command line variable " + name + " not found in the configuration. Ignoring.");
+            } else {
+                String value = entry.getValue().getLeft();
+                String parser = entry.getValue().getRight();
+                Sort sort = compiledDef.configurationVariableDefaultSorts.getOrDefault(configVarName, compiledDef.programStartSymbol);
+                K configVar = externalParse(parser, value, sort, Source.apply("<command line: -c" + name + ">"), compiledDef);
+                output.put(KToken(configVarName, Sorts.KConfigVar()), configVar);
+            }
         }
-        if (compiledDef.kompiledDefinition.mainModule().allSorts().contains(Sorts.String())) {
+        if (compiledDef.kompiledDefinition.mainModule().allSorts().contains(Sorts.String()))
```

Comment:
please don't remove braces like this. I know it's not strictly necessary since there's only one statement inside the braces, but it's still bad code style.